omaha/core/core.cc

```diff
@@ -277,9 +277,34 @@ HRESULT Core::DoMain(bool is_system, bool is_crash_handler_enabled) {
     return hr;
   }
 
-  std::unique_ptr<Scheduler> scheduler(new Scheduler(*this));
-  hr = scheduler->Initialize();
+  auto scheduler = std::make_unique<Scheduler>();
```

Comment:
It would be nice to encapsulate these scheduling details somehow. Before the change, the code looked simple: new, then initialize. After the change, there is a lot more that has spilled out into core.cc

Reply:
Moved it to `Core::InitializeScheduler` method. Hope this works!